week4/ex5.c

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ *	Author: Elnama Mammadova
+ *
+ *	Problem:
+ *
+ *	There are two runners that run on a road. Write a program that given the initial
+ *	coordinates of the runners, and the speeds (size of step) prints the coordinate and time
+ *	taken where they will meet or pass each other OR prints NEVER in case, they are running
+ *	from each other, or they are running towards the same direction but the one behind has not enough speed.
+ *
+ *
+ *	Solution is provided as comments in the code for better understanding.
+ *
+ */
+
+#include <stdio.h>
+
+int main() {
+	int x1, v1, x2, v2;
+	int time = 0;
+
+	// Input for initial positions and speeds
+	scanf("%d %d", &x1, &v1);
+	scanf("%d %d", &x2, &v2);
+
+	// Check for cases where they will never meet
+	if ((x1 < x2 && v1 <= v2) || (x1 > x2 && v1 >= v2) ||
+		(x1 < x2 && v1 < 0 && v2 > 0) || (x1 > x2 && v1 > 0 && v2 < 0)) {
```

Comment:
Here I believe the above two conditions already cover the later two.

For instance: if   (x1 < x2 && v1 <= v2) is true that means (x1 < x2 && v1 < 0 && v2 > 0) is also true, because
v1 < 0 && v2 > 0   => v1 <= v2.

So the entire if case can be rewritten shorter: if ((x1 < x2 && v1 <= v2) || (x1 > x2 && v1 >= v2)). Don't you think?